pyresample/bilinear/__init__.py

```diff
@@ -210,366 +200,772 @@ def get_bil_info(source_geo_def, target_area_def, radius=50e3, neighbours=32,
+    pt_3.
+
+    """
+    pt_1, pt_2, pt_3, pt_4 = corner_points
     # Pairwise longitudal separations between reference points
     x_21 = pt_2[:, 0] - pt_1[:, 0]
     x_31 = pt_3[:, 0] - pt_1[:, 0]
+    x_42 = pt_4[:, 0] - pt_2[:, 0]
 
     # Pairwise latitudal separations between reference points
     y_21 = pt_2[:, 1] - pt_1[:, 1]
     y_31 = pt_3[:, 1] - pt_1[:, 1]
+    y_42 = pt_4[:, 1] - pt_2[:, 1]
 
-    t__ = (x_21 * (out_y - pt_1[:, 1]) - y_21 * (out_x - pt_1[:, 0])) / \
-          (x_21 * y_31 - y_21 * x_31)
-    with np.errstate(invalid='ignore'):
-        idxs = (t__ < 0.) | (t__ > 1.)
-    t__[idxs] = np.nan
+    a__ = x_31 * y_42 - y_31 * x_42
+    b__ = out_y * (x_42 - x_31) - out_x * (y_42 - y_31) + \
+        x_31 * pt_2[:, 1] - y_31 * pt_2[:, 0] + \
+        y_42 * pt_1[:, 0] - x_42 * pt_1[:, 1]
+    c__ = out_y * x_21 - out_x * y_21 + pt_1[:, 0] * pt_2[:, 1] - \
+        pt_2[:, 0] * pt_1[:, 1]
 
-    s__ = (out_x - pt_1[:, 0] + x_31 * t__) / x_21
+    return a__, b__, c__
 
-    with np.errstate(invalid='ignore'):
-        idxs = (s__ < 0.) | (s__ > 1.)
-    s__[idxs] = np.nan
 
-    return t__, s__
+def _solve_another_fractional_distance(f__, y_corners, out_y):
+    """Solve parameter t__ from s__, or vice versa.
 
+    For solving s__, switch order of y_2 and y_3.
+    """
+    y_1, y_2, y_3, y_4 = y_corners
 
-def _solve_another_fractional_distance(f__, y_1, y_2, y_3, y_4, out_y):
-    """Solve parameter t__ from s__, or vice versa.  For solving s__,
-    switch order of y_2 and y_3."""
     y_21 = y_2 - y_1
     y_43 = y_4 - y_3
 
-    with np.errstate(divide='ignore'):
-        g__ = ((out_y - y_1 - y_21 * f__) /
-               (y_3 + y_43 * f__ - y_1 - y_21 * f__))
+    g__ = ((out_y - y_1 - y_21 * f__) /
+           (y_3 + y_43 * f__ - y_1 - y_21 * f__))
 
     # Limit values to interval [0, 1]
-    with np.errstate(invalid='ignore'):
-        idxs = (g__ < 0) | (g__ > 1)
-    g__[idxs] = np.nan
+    g__ = np.where(
+        find_indices_outside_min_and_max(g__, 0, 1),
+        np.nan, g__)
 
     return g__
 
 
-def _calc_abc(pt_1, pt_2, pt_3, pt_4, out_y, out_x):
-    """Calculate coefficients for quadratic equation for
-    _get_ts_irregular() and _get_ts_uprights().  For _get_ts_uprights
-    switch order of pt_2 and pt_3.
-    """
+def _update_fractional_distances(func, t__, s__, corner_points, out_x, out_y):
+    idxs = np.ravel(np.isnan(t__) | np.isnan(s__))
+    if np.any(idxs):
+        new_values = func(corner_points, out_y, out_x)
+        updated_t_and_s = _np_where_for_multiple_arrays(idxs, new_values, (t__, s__))
+        t__, s__ = _invalid_s_and_t_to_nan(*updated_t_and_s)
+    return t__, s__
+
+
+def _get_fractional_distances_uprights_parallel(corner_points, out_y, out_x):
+    """Get parameters for the case where uprights are parallel."""
+    pt_1, pt_2, pt_3, pt_4 = corner_points
+    # Get the valid roots from interval [0, 1]
+    corner_points = (pt_1, pt_3, pt_2, pt_4)
```

Comment:
what?

Reply:
The following function call needs the corners in different order than normal. It's part of the algorithm the resampler is based on, and I've not dug in it so can't say why.

Reply:
oh, sorry, I missed the different order. Maybe a comment could help for that.

Reply:
Explanation added